Zend/zend_vm_def.h

```diff
@@ -9549,6 +9549,71 @@ ZEND_VM_HANDLER(202, ZEND_CALLABLE_CONVERT, UNUSED, UNUSED)
 	ZEND_VM_NEXT_OPCODE();
 }
 
+ZEND_VM_HANDLER(204, ZEND_FRAMELESS_ICALL_0, UNUSED, UNUSED)
+{
+	USE_OPLINE
+	SAVE_OPLINE();
+	zend_frameless_function_0 function = zend_frameless_function_0_handlers[opline->extended_value];
+	function(EX_VAR(opline->result.var));
+	if (EG(exception)) {
+		ZVAL_NULL(EX_VAR(opline->result.var));
+		HANDLE_EXCEPTION();
+	}
+	ZEND_VM_NEXT_OPCODE();
+}
+
+ZEND_VM_HANDLER(205, ZEND_FRAMELESS_ICALL_1, CONST|TMP|VAR|CV, UNUSED)
+{
+	USE_OPLINE
+	SAVE_OPLINE();
+	zend_frameless_function_1 function = zend_frameless_function_1_handlers[opline->extended_value];
+	function(EX_VAR(opline->result.var), GET_OP1_ZVAL_PTR_DEREF(BP_VAR_R));
+	FREE_OP1();
+	if (EG(exception)) {
+		ZVAL_NULL(EX_VAR(opline->result.var));
+		HANDLE_EXCEPTION();
+	}
+	ZEND_VM_NEXT_OPCODE();
+}
+
+ZEND_VM_HANDLER(206, ZEND_FRAMELESS_ICALL_2, CONST|TMP|VAR|CV, CONST|TMP|VAR|CV)
+{
+	USE_OPLINE
+	SAVE_OPLINE();
+	zend_frameless_function_2 function = zend_frameless_function_2_handlers[opline->extended_value];
+	function(
+		EX_VAR(opline->result.var),
+		GET_OP1_ZVAL_PTR_DEREF(BP_VAR_R),
+		GET_OP2_ZVAL_PTR_DEREF(BP_VAR_R));
```

Comment:
You don't check arguments for IS_UNDEF. How they are handled?

Reply:
`ZEND_FRAMELESS_ICALL_2` doesn't accept `IS_UNDEF`. What needs to be handled?

Reply:
I mean undefined CV variables.

Reply:
Do you mean checking for `EG(exception)` after `GET_OPx_ZVAL_PTR_DEREF`? That's indeed missing (as an error handler with an exception on `SEND` with an undefined var would prevent the function from being called).

Reply:
I mean the following code.
```php
<?php
class_exists($undefined_var);
```
I actually forgot that ``GET_OP1_ZVAL_PTR_DEREF()`` checks for IS_UNDEF itself. :)
Anyway:
- after IS_UNDEF check we need ``EG(excepion)`` check
- when ``zend_error(E_WARNING, "Undefined variable $%s")`` is called, ``get_active_function_or_method_name()`` would return ``"class_exists"`` instead of current method name.

Reply:
> when zend_error(E_WARNING, "Undefined variable $%s") is called, get_active_function_or_method_name() would return "class_exists" instead of current method name.

Hmm, indeed. Good catch.